### PR TITLE
Fulcrum Logging requirements

### PR DESCRIPTION
**Pull request by adrianhall**
Added default logging requirements to the general guidelines.

### CODE REVIEW

File: docs/general/implementation.md [hunk 1]
```diff
@@ -141,12 +141,101 @@ Client libraries must support robust logging mechanisms so that the consumer can
...
 
+### Default Logging
+
+Logging for consumers is generally opt-in and aids the consumer in diagnosis of problems with their code and potential service issues.  However, there are cases where we require default logging, primarily in customer support situations.  The main goal of default logging is to improve customer satisfaction with the service when used in combination with the Azure SDK.  If a consumer requests support from Microsoft for a service operation via the client library, certain information is required in order to diagnose the issue.  We want to ensure that information is always available.  The Azure SDK supports several languages on several operating systems and within several app models.  As such, logging is not always available nor is it advisable in all situations.
+
+{% include requirement/MUST id="fulcrum-logging-required" %} enable default logging when the `TEMP` environment variable exists and file system APIs are available.  The `TEMP` environment variable must point to a writable directory on the filesystem.  In cases where there is no file system, default logging is not available.
```
Comment (joshfree) [inline docs/general/implementation.md]: TEMP is very windows-centric.  TMPDIR should be used for Linux and Mac

File: docs/general/implementation.md [hunk 2]
```diff
@@ -141,12 +141,101 @@ Client libraries must support robust logging mechanisms so that the consumer can
...
+{% include requirement/MUSTNOT id="fulcrum-logging-prohibited" %} enable default logging if the platform does not allow user access to the file system.  For example, mobile environments generally have a file system that is not user accessible.
+
+{% include requirement/MUST id="fulcrum-logging-optout" %} allow the consumer to opt out of default logging. 
+
+The consumer should be able to opt out as follows:
+
+* Globally by setting a `AZURE_DISABLE_DEFAULT_LOGGING` environment variable.
+* On a per-client basis by setting a `disableDefaultLogging` client option to true.
+
+{% include requirement/MUST id="fulcrum-logging-where" %} write logs to a sub-directory of the `TEMP` folder named `azurelogs`.  Logs for all service clients should write to the same log file as this allows CSS to identify cross-service issues more easily.
```
Comment (joshfree) [inline docs/general/implementation.md]: should include TMPDIR

Comment (adrianhall) [reply docs/general/implementation.md]: Agreed.  I've added that requirement.

**Comment by adrianhall** [comment]
Per the architectural board, this change is rejected as a guideline update.  We will engage directly with the Fulcrum team to determine what the underlying requirements for supportability are and adjust as needed.  The actual data requirements (such as slow links) has been moved to the other logging update PR.

**Pull request closed without merging**